pkg/kubelet/preemption/preemption_test.go

```diff
@@ -0,0 +1,480 @@
+	}
+	for n := 0; n < t.N; n++ {
+		getPodsToPreempt(inputPods, admissionRequirementList([]*admissionRequirement{
+			{
+				resourceName: v1.ResourceCPU,
+				quantity:     parseCPUToInt64("110m"),
+			}}))
+	}
+}
+
+func TestGetPodsToPreempt(t *testing.T) {
+	type testRun struct {
+		testName              string
+		inputPods             []*v1.Pod
+		insufficientResources admissionRequirementList
+		expectErr             bool
+		expectedOutput        []*v1.Pod
+	}
+	allPods := getTestPods()
+	runs := []testRun{
+		{
+			testName:              "no requirements",
+			inputPods:             []*v1.Pod{},
+			insufficientResources: getAdmissionRequirementList(0, 0, 0),
+			expectErr:             false,
+			expectedOutput:        []*v1.Pod{},
+		},
+		{
+			testName:              "no pods",
+			inputPods:             []*v1.Pod{},
+			insufficientResources: getAdmissionRequirementList(0, 0, 1),
+			expectErr:             true,
+			expectedOutput:        nil,
+		},
+		{
+			testName:              "equal pods and resources requirements",
+			inputPods:             []*v1.Pod{allPods[burstable]},
+			insufficientResources: getAdmissionRequirementList(100, 100, 1),
+			expectErr:             false,
+			expectedOutput:        []*v1.Pod{allPods[burstable]},
+		},
+		{
+			testName:              "higer requirements than pod requests",
+			inputPods:             []*v1.Pod{allPods[burstable]},
+			insufficientResources: getAdmissionRequirementList(200, 200, 2),
+			expectErr:             true,
+			expectedOutput:        nil,
+		},
+		{
+			testName:              "choose between bestEffort and burstable",
+			inputPods:             []*v1.Pod{allPods[burstable], allPods[bestEffort]},
+			insufficientResources: getAdmissionRequirementList(0, 0, 1),
+			expectErr:             false,
+			expectedOutput:        []*v1.Pod{allPods[bestEffort]},
+		},
+		{
+			testName:              "choose between burstable and guaranteed",
+			inputPods:             []*v1.Pod{allPods[burstable], allPods[guaranteed]},
+			insufficientResources: getAdmissionRequirementList(0, 0, 1),
+			expectErr:             false,
+			expectedOutput:        []*v1.Pod{allPods[burstable]},
+		},
+		{
+			testName:              "choose lower request burstable if it meets requirements",
+			inputPods:             []*v1.Pod{allPods[highRequestBurstable], allPods[burstable]},
+			insufficientResources: getAdmissionRequirementList(100, 100, 0),
+			expectErr:             false,
+			expectedOutput:        []*v1.Pod{allPods[burstable]},
+		},
+		{
+			testName:              "choose higher request burstable if lower does not meet requirements",
+			inputPods:             []*v1.Pod{allPods[burstable], allPods[highRequestBurstable]},
+			insufficientResources: getAdmissionRequirementList(150, 150, 0),
+			expectErr:             false,
+			expectedOutput:        []*v1.Pod{allPods[highRequestBurstable]},
+		},
+		{
+			testName:              "multiple pods required",
```

Comment:
Can you add bestEffort pods to this mix and check to see that only burstable gets evicted?